### ISSUE TITLE
[FR] LSP Integration

### ISSUE THREAD

**Issue by cgardner** [renerocksai/telekasten.nvim#97]
**Please confirm**
- [x] I am running the latest version of this plugin

**Is your feature request related to a problem? Please describe.**
When working with notes using telekasten, I need to remember exactly how I spelled all of my tags and notes

**Describe the solution you'd like**
LSP integration.  The ability to auto complete tags and notes for linking

**Describe alternatives you've considered**
The built in `insert_link` command works, but it would be nice to enter `[[thi` and be able to suggest notes that start with `thi`

**Additional context**

**Comment by reisnobre** [comment]
That would involve the creation af an whole new LSP right? It seems pretty intense.
I really would like to learn how to build something like that.

**Comment by tami5** [comment]
> That would involve the creation af an whole new LSP right? It seems pretty intense.

Nope, not with [jose-elias-alvarez/null-ls.nvim](https://github.com/jose-elias-alvarez/null-ls.nvim) being a thing. there is a completion support, also hovers, also code actions, 😆 @jose-elias-alvarez you rock!

**Comment by reisnobre** [comment]
True true, I use it everyday as part of LunarVim/LunarVim. It's so seamless that I even forgot tha it could be used like that.

**Comment by zalegrala** [comment]
This would be amazing.

**Comment by lambtho12** [comment]
This is aleady mentionned in [the BACKLOG](https://github.com/renerocksai/telekasten.nvim/blob/ea4d30de0060037de6dd34073ff0de51bd1305cf/BACKLOG.md)

LSP support was also disscussed in #23

**Comment by andy4thehuynh** [comment]
@lambtho12 I've paged through the backlog and #23 and unsure if support for this feature exists yet. Please let me know if completion support for tags exists and if there's a Telekasten command for it. Thanks again for all your work on this project!

**Comment by lambtho12** [comment]
@andy4thehuynh There is currently no auto-completion, nor LSP in telekasten. @renerocksai and myself are quite busy at the moment so the development on telekasten has been rather slowed down these lasts months.

LSP is a major chunk of work. I do not know when we will start working on that. That being said, I still maintain actively the project and review/merge PRs. If you (or anyone else) want to start working on that, I will be more than happy to review it!

The only current implementation of tags is with `show_tags`. You can then use the pickers actions `<C-y>` to yank the tag or `<C-i>` to insert the tag.

**Comment by BonaBeavis** [comment]
You can use [marksman](https://github.com/artempyanykh/marksman). This gives autocompletion for links navigation to links and preview of links.  Adding autocompletion for tags in marksman would be relatively easy. 

Another avenue could be adding a source to [cmp](https://github.com/hrsh7th/nvim-cmp)

### Hover

![image](https://user-images.githubusercontent.com/3330855/203498982-ccb1cbbb-1a91-4171-a2bf-96fec3e6b93e.png)

### Link completion

![image](https://user-images.githubusercontent.com/3330855/203499138-7425d44c-1c50-4473-b722-f34a2e0024b5.png)

**Comment by thw26** [comment]
This is also implemented in another similar nvim plugin released under Apache 2.0:

https://github.com/epwalsh/obsidian.nvim/blob/main/lua/cmp_obsidian.lua

**Comment by Sleepful** [comment]
Since we are on the topic, another plugin that is very nice for markdown files is https://github.com/dkarter/bullets.vim
I like to use this one with `marksman` lsp and it gives a very automated typing experience.
